packages/gatsby-plugin-netlify-cache/src/gatsby-node.js

```diff
@@ -0,0 +1,55 @@
+const { resolve } = require(`path`)
+
+const { ensureDir, readdir, copy } = require(`fs-extra`)
+
+function calculateDirs (store) {
+  const program = store.getState().program
+
+  const staticDir = resolve(program.directory, `public`, `static`)
+  const cacheDir = process.env.NETLIFY_BUILD_BASE
+    ? resolve(process.env.NETLIFY_BUILD_BASE, `cache`, `.gatsby-static-files`)
```

Comment:
does this need to be `.cache` instead of `cache`?

i'm looking at their build image here: https://github.com/netlify/build-image/blob/master/run-build-functions.sh#L24

Reply:
> does this need to be .cache instead of cache?

eh, probably not as that appears to be [pip's](https://pypi.org/project/pip/) cache. 

some more observations below - hope it helps!

logging the values of:

```
NETLIFY_BUILD_BASE
NETLIFY_CACHE_DIR
NETLIFY_REPO_DIR
```

as found [here](https://github.com/netlify/build-image/blob/master/run-build.sh#L7), the only one that is set is `NETLIFY_BUILD_BASE` - its value was: `/opt/build`

reading thru their build code, it appears that they are pretty conservative in what they actually allow to be cached and what they restore:

cached dirs - [see here](https://github.com/netlify/build-image/blob/master/run-build-functions.sh#L493)
restored dirs - [see here](https://github.com/netlify/build-image/blob/master/run-build-functions.sh#L143) and grep for `restore_home_cache` and `restore_cwd_cache`. 

lastly, i tried hardcoding the expected `cacheDir` path.  i tried all of the following without any luck:

`/opt/build/cache/.gatsby-static-files`
`/opt/buildhome/cache/.gatsby-static-files`
`/opt/build/repo/.gatsby-static-files`
`/opt/build/repo/node_modules/.gatsby-static-files`

so i'm not sure how to get things working at this point.

Reply:
ok, i got it to work by manually re-triggering a deploy via netlify's web admin ui and checking the "Clear build cache" box in the confirm dialog.

of course, you can only trigger a deploy for the "production" branch (`master` in my case) which is why the deploy preview builds weren't working.

files are now saved and restored but given what it took to get here i'm wondering how reliable this will prove to be? how will new files be added to the cache in the future? will i always need to trigger a deploy and clear the build cache for changes (additions primarily) to be detected?

🤷‍♂️

anyway, thanks for putting this together 🙏